src/freenet/clients/http/UserAlertsToadlet.java

```diff
@@ -47,7 +50,16 @@ public void handleMethodPOST(URI uri, HTTPRequest request, ToadletContext ctx) t
 			int userAlertHashCode = request.getIntPart("disable", -1);
 			ctx.getAlertManager().dismissAlert(userAlertHashCode);
 		}
-		headers.put("Location", ".");
+		
+		
+		String redirect;
+		try {
+			redirect = request.getPartAsStringThrowing("redirectToAfterDisable", 1024);
+		} catch (SizeLimitExceededException | NoSuchElementException e) {
```

Comment:
Argh that's my crap API, sorry. Wrote that many years ago when I was still stupid enough to just chose exceptions whose names "sound about right", without considering their JavaDoc.
I'm ashamed that I ever was that stupid!

People previously complained that my pull requests are too large, so I will resolve it in a different one as the function wasn't added by this PR anyway.
I've [filed a bug](https://bugs.freenetproject.org/view.php?id=6908) and assigned it to myself.

With regards to using ```IllegalArgumentException```:
Whether or not exceptions are thrown does not signal faulty arguments but whether the remote HTTP request contains suitable data.
So if we used ```IllegalArgumentException```, it could happen that for a specific constant choice of arguments it is sometimes thrown and sometimes not thrown, with the same arguments. But a fixed set of arguments either should be legal or not I suppose?
Also callers could need the distinction between the two situations of the remote user giving too much data or no data at all. Too much data = we need to tell the user they entered too much text in a textbox; No data = we need to tell the user that they must enter something.

Considering that, do you have a better idea than ```IllegalArgumentException```?
If not, I'll just create custom exception classes ```PartTooLargeException``` / ```NoSuchPartException```.